src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<OrderResponseDto, Order> dtoResponseMapper;
+    private final UserService userService;
+
+    public OrderController(OrderService orderService,
+                           ShoppingCartService shoppingCartService,
+                           DtoResponseMapper<OrderResponseDto, Order> dtoResponseMapper,
+                           UserService userService) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.dtoResponseMapper = dtoResponseMapper;
+        this.userService = userService;
+    }
+
+    @PostMapping("/complete{userId}")
+    public OrderResponseDto completeOrder(@PathVariable Long userId) {
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(userService.get(userId));
+        return dtoResponseMapper.toDto(orderService.completeOrder(shoppingCart));
+    }
+
+    @GetMapping("{userId}")
+    public List<OrderResponseDto> getOrderHistory(@PathVariable Long userId) {

```

Comment:
According the task description "Get orders history for user - GET: /orders?userId"
```suggestion
   @GetMapping
    public List<OrderResponseDto> getOrdersHistory(@RequestParam Long userId) {

```